From: Riley <oreo4455@gmail.com>
Subject: [PATCH] ASoC: codecs: wcd9335: add decimator volume controls

---
diff --git a/sound/soc/codecs/wcd9335.c b/sound/soc/codecs/wcd9335.c
index 4515c82e5..990865649 100644
--- a/sound/soc/codecs/wcd9335.c
+++ b/sound/soc/codecs/wcd9335.c
@@ -2296,6 +2296,24 @@ static int wcd9335_rx_hph_mode_put(struct snd_kcontrol *kc,
 }
 
 static const struct snd_kcontrol_new wcd9335_snd_controls[] = {
+	SOC_SINGLE_S8_TLV("DEC0 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 0,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC1 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 1,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC2 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 2,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC3 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 3,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC4 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 4,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC5 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 5,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC6 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 6,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC7 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 7,
+			-84, 40, digital_gain),
+	SOC_SINGLE_S8_TLV("DEC8 Volume", WCD9335_CDC_TX0_TX_VOL_CTL + 16 * 8,
+			-84, 40, digital_gain),
 	/* -84dB min - 40dB max */
 	SOC_SINGLE_S8_TLV("RX0 Digital Volume", WCD9335_CDC_RX0_RX_VOL_CTL,
 			-84, 40, digital_gain),
